src/__snapshots__/ErrorMessage.test.tsx.snap

```diff
@@ -8,7 +8,21 @@ exports[`ErrorMessage should render correctly with flat errors 1`] = `
 </DocumentFragment>
 `;
 
-exports[`ErrorMessage should render correctly with flat errors and as with component and className and render 1`] = `
+exports[`ErrorMessage should render correctly with flat errors and as with component 1`] = `
+<DocumentFragment>
+  <div>
+    flat
+  </div>
+</DocumentFragment>
+`;
+
+exports[`ErrorMessage should render correctly with flat errors and as with component and render 1`] = `
+<DocumentFragment>
+  flat
+</DocumentFragment>
```

Comment:
This looks incorrect.  According to the README, the `as` should take precedence, but that seems to only apply to elements provided in `as`, not components.

Reply:
@lanVS oh... you are right!
We need to fix this.

Reply:
Do you need me to try to work on it in this PR, or can that maybe be a separate effort?

Reply:
Let's merge this once 👍🏻 
If you are willing to help, we would like you to create another PR 🙏🏻

Reply:
I'll see what I can do, but TBH I'm stretched a bit thin at the moment.